src/simtools/testing/sim_telarray_metadata.py

```diff
@@ -0,0 +1,164 @@
+            else:
+                value = metadata[sim_telarray_name]
+                value = (int)(value) if value.isnumeric() else value
+
+            _logger.debug(
+                f"Parameter {param} in sim_telarray file: {value}, "
+                f"in model: {model.parameters[param]['value']}"
+            )
+            if not is_equal(value, model.parameters[param]["value"], parameter_type):
+                invalid_parameter_list.append(
+                    f"Parameter {param} mismatch between sim_telarray file: {value}, "
+                    f"and model: {model.parameters[param]['value']}"
+                )
+
+    return invalid_parameter_list
+
+
+def _sim_telarray_name_from_parameter_name(parameter_name):
+    """Return sim_telarray parameter name. Some specific fine tuning."""
+    # parameters like "reference_point_latitude"
+    sim_telarray_name = parameter_name.replace("reference_point_", "")
+
+    if sim_telarray_name == "altitude":
+        return "corsika_observation_level"
+    if sim_telarray_name == "array_triggers":
+        return None
+
+    return sim_telarray_name
+
+
+def is_equal(value1, value2, value_type):
+    """
+    Check if two values are equal based on their type.
+
+    The complexity of this function reflects the complexity of the sim_telarray
+    metadata output.
+
+    Parameters
+    ----------
+    value1: any
+        First value to compare.
+    value2: any
+        Second value to compare.
+    value_type: str
+        Type of the values ('string', 'dict', etc.).
+
+    Returns
+    -------
+    bool
+        True if the values are equal, False otherwise.
+    """
+    value1 = value1[0] if isinstance(value1, tuple) else value1
+    value2 = value2[0] if isinstance(value2, tuple) else value2
+    if value1 is None or value2 is None:
+        if value1 in ("none", None) and value2 in ("none", None):
+            return True
+    if value_type == "string":
+        return str(value1).strip() == str(value2).strip()
+    if value_type == "dict":
+        return value1 == value2
+    if value_type == "boolean":
+        return bool(value1) == bool(value2)
+    return _is_equal_floats_or_ints(value1, value2)
+
+
+def _is_equal_floats_or_ints(value1, value2):
+    """Check if floats and ints are equal."""
+    if isinstance(value1, np.ndarray | list) and isinstance(value2, np.ndarray | list):
+        return bool(np.allclose(np.array(value1), np.array(value2), rtol=1e-10))
+    if isinstance(value1, list) and isinstance(value2, float | int | np.integer | np.floating):
+        if all(x == value1[0] for x in value1):
+            return bool(np.isclose(float(value1[0]), float(value2), rtol=1e-10))
+    if isinstance(value2, list) and isinstance(value1, float | int | np.integer | np.floating):
+        if all(x == value2[0] for x in value2):
+            return bool(np.isclose(float(value1), float(value2[0]), rtol=1e-10))
+    return bool(np.isclose(float(value1), float(value2), rtol=1e-10))
```

Comment:
We could consider moving these two functions to general utilities (in particular if the rtol is made an argument).

Reply:
Yes and no - this is a bit fine tuned to the model parameters (e.g., the conversion from json types to python types), the "none", "None" conversion, the possibility for tuble input) and the style we store the data (numpy arrays). 

Do you have another application in mind where this might be useful?

Reply:
No, no application, it just looked like a fairly general function (in particular the second one). I really don't mind it staying here though, so I am resolving this comment.